src/alasqlparser.jison

```diff
@@ -1394,18 +1429,36 @@ FuncValue
 		{ $$ = new yy.FuncValue({ funcid: 'REPLACE', args:$3 }) }
 	| CURRENT_DATE LPAR RPAR
 		{ $$ = new yy.FuncValue({ funcid: $1 }) }
-	| DATEADD LPAR Literal COMMA Expression COMMA Expression RPAR
-		{ $$ = new yy.FuncValue({ funcid: 'DATEADD', args:[new yy.StringValue({value:$3}),$5,$7]}) }
+	| CURRENT_TIMESTAMP LPAR RPAR
+		{ $$ = new yy.FuncValue({ funcid: $1 }) }
 	| DATEADD LPAR STRING COMMA Expression COMMA Expression RPAR
 		{ $$ = new yy.FuncValue({ funcid: 'DATEADD', args:[$3,$5,$7]}) }
-	| DATEDIFF LPAR Literal COMMA Expression COMMA Expression RPAR
-		{ $$ = new yy.FuncValue({ funcid: 'DATEDIFF', args:[new yy.StringValue({value:$3}),$5,$7]}) }
+	| DATEADD LPAR IntervalLiteral COMMA Expression COMMA Expression RPAR
+		{ $$ = new yy.FuncValue({ funcid: 'DATEADD', args:[new yy.StringValue({value:$3}),$5,$7]}) }
 	| DATEDIFF LPAR STRING COMMA Expression COMMA Expression RPAR
 		{ $$ = new yy.FuncValue({ funcid: 'DATEDIFF', args:[$3,$5,$7]}) }
+	| DATEDIFF LPAR IntervalLiteral COMMA Expression COMMA Expression RPAR
+		{ $$ = new yy.FuncValue({ funcid: 'DATEDIFF', args:[new yy.StringValue({value:$3}),$5,$7]}) }
+	| DATE_ADD LPAR Expression COMMA Expression RPAR
+	    /* e.g. DATE_ADD(NOW(), 24 * 3600 * 1000) */
+		{ $$ = new yy.FuncValue({ funcid: 'DATE_ADD', args:[$3, $5] }) }
+	| DATE_ADD LPAR Expression COMMA INTERVAL Expression IntervalLiteral RPAR
+	    /* e.g. DATE_ADD(NOW(), INTERVAL 10 day) */
+		{ $$ = new yy.FuncValue({ funcid: 'DATE_ADD', args:[$3, new yy.FuncValue({ funcid: 'INTERVAL', args:[$6,new yy.StringValue({value:($7).toLowerCase()})]})] }) }
+	| DATE_SUB LPAR Expression COMMA Expression RPAR
+	    /* e.g. DATE_SUB(NOW(), 24 * 3600 * 1000) */
+		{ $$ = new yy.FuncValue({ funcid: 'DATE_SUB', args:[$3, $5]}) }
+	| DATE_SUB LPAR Expression COMMA INTERVAL Expression IntervalLiteral RPAR
+		/* e.g. DATE_SUB(NOW(), INTERVAL 10 day) */
+		{ $$ = new yy.FuncValue({ funcid: 'DATE_SUB', args:[$3, new yy.FuncValue({ funcid: 'INTERVAL', args:[$6,new yy.StringValue({value:($7).toLowerCase()})]}) ]}) }
+	| (YEAR|MONTH|DAY|DAYOFWEEK|HOUR|MINUTE|SECOND) LPAR Expression RPAR

```

Comment:
Very nice. I was not aware this notation was possible.

Reply:
I added the other aliases as well.